index/types.go

```diff
@@ -42,6 +40,17 @@ type Index interface {
 	Close() error
 }
 
+// Writer is used to insert documents into an index.
+type Writer interface {
+	// Insert inserts the given document into the index and returns its ID. The document
+	// is guaranteed to be searchable once the Insert method returns.
+	Insert(d doc.Document) ([]byte, error)
+
+	// InsertBatch inserts a batch of metrics into the index. The documents are guaranteed
+	// to be searchable all at once when the Batch method returns.
```

Comment:
Maybe document the types of errors this can return so it's not a complete surprise when people are downcasting.

Reply:
Definitely, good idea!